three-cards-divination/src/test/java/com/cyber/mysticism/tarot/service/UserServiceTest.java

```diff
@@ -0,0 +1,124 @@
+package com.cyber.mysticism.tarot.service;
+
+import com.cyber.mysticism.tarot.model.TarotUser;
+import com.cyber.mysticism.tarot.repository.UserRepository;
+import com.cyber.mysticism.tarot.service.exceptions.UserAlreadyExistsException;
+import com.cyber.mysticism.tarot.service.exceptions.UserNotFoundException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+class UserServiceTest {
+
+    @Mock
+    private UserRepository userRepository;
+    @InjectMocks
+    private UserService userService;
+
+    private TarotUser tarotUser;
+
+    @BeforeEach
+    void setup() {
+        tarotUser = new TarotUser();
+        tarotUser.setUsername("user");
+        tarotUser.setEmail("email");
+    }
```

Comment:
**Suggestion:** The `setup` method is setting up a `TarotUser` object for every test method in the class. However, not all test methods require this object. Consider moving the `TarotUser` setup to the individual test methods that require it. This will make the tests more isolated and easier to understand. [maintainability]
```suggestion
    // Remove the setup method and move the TarotUser setup to the individual test methods
```